src/Agent.Worker/Handlers/NodeHandler.cs

```diff
@@ -73,15 +74,28 @@ public sealed class NodeHandler : Handler, INodeHandler
                 processInvoker.ErrorDataReceived += OnDataReceived;
                 string node = Path.Combine(
                     IOUtil.GetExternalsPath(),
-                    "node",
+                    ExecutionContext.Variables.Agent_UseNode5 ? "node-5.10.1" : "node",
                     "bin",
                     $"node{IOUtil.ExeExtension}");
 
-                // Format the arguments passed to node.
-                // 1) Wrap the script file path in double quotes.
-                // 2) Escape double quotes within the script file path. Double-quote is a valid
+                var arguments = new List<string>();
+                if (!ExecutionContext.Variables.Agent_UseNode5)
+                {
+                    // Set --no-warnings to help preserve back compat. The handler previously used node version v5.10.1.
+                    // When the handler was changed to use node v6.10.3, node can emit certain warnings over stderr.
+                    // Passing the --no-warnings flag disables the warnings.
+                    //
+                    // Note, passing the flag --no-warnings will not disable the warnings for downstream node processes.
+                    // If customers run into issues with this problem, then they can set the env var NODE_NO_WARNINGS=1.
+                    // The agent should not set this env var, since it is unknown whether the task author desires warning
+                    // suppression for downstream processes. For example, if the custom task uses the node installer task
+                    // to run ad hoc customer scripts, then it should not set the env var.
+                    arguments.Add("--no-warnings");
```

Comment:
Also I don't think the env var is supported on win in node 6 (command line flag is supported), so another reason we shouldnt set the env var.